sample-resource-plugin/src/main/java/org/opensearch/sample/resource/actions/transport/RevokeResourceAccessTransportAction.java

```diff
@@ -39,16 +44,12 @@ public RevokeResourceAccessTransportAction(TransportService transportService, Ac
     @Override
     protected void doExecute(Task task, RevokeResourceAccessRequest request, ActionListener<RevokeResourceAccessResponse> listener) {
         ResourceSharingClient resourceSharingClient = ResourceSharingClientAccessor.getInstance().getResourceSharingClient();
-        resourceSharingClient.revoke(
-            request.getResourceId(),
-            RESOURCE_INDEX_NAME,
-            request.getEntitiesToRevoke(),
-            ActionListener.wrap(success -> {
-                RevokeResourceAccessResponse response = new RevokeResourceAccessResponse(success.getShareWith());
-                log.debug("Revoked resource access: {}", response.toString());
-                listener.onResponse(response);
-            }, listener::onFailure)
-        );
+        ShareWith target = new ShareWith(Map.of(PLACE_HOLDER, new Recipients(request.getEntitiesToRevoke())));
```

Comment:
Does this mean that the plugins must now pass the name PLACE_HOLDER?

Reply:
yes, they would pass the whole ShareWith data structure which is is organized by access level and then by entities to removed organized by category (users, roles)

If we enforce uniqueness (i.e. user can be add accessLevelA or accessLevelB but not both) then it wouldn't be necessary to include accessLevel when revoking. It would just be necessary to supply the username or rolename being revoked.

Reply:
okay.

Enforcing uniqueness becomes cumbersome as number of calls may increase, as each user will have to be verified and then the correct permissions will have to be chosen.

Let's keep it open to accept access level as input.